### PR TITLE
[fix] Handle cancellation of core.InitAsync()

### LINKED ISSUE #135: Cancelling core configuration in OOBE causes crash
### Describe the bug

Attempting to cancel the setup of a new core during the OOBE causes the app to crash with a `TaskCanceledException`.

### Affected area

- [ ] Strix Music SDK
- [X] Strix Music App

### Regression

_No response_

### Steps to reproduce

```text
1. Open a fresh install of Strix Music
2. In the Services tab of the OOBE, add a OneDrive core
3. Click "OK"
4. Click "Cancel"
5. Observe app crash
```

### Visual repro steps

_No response_

### Expected behavior

The OOBE should gracefully cancel setup of the selected core.

### Additional context

_No response_

### Help us help you

Yes, but only if others can assist.

### PR DESCRIPTION
**Pull request by yoshiask**
## Overview

Closes #135

Currently, `AppLoadingView` calls `core.InitAsync()` without considering whether the initialization or configuration might be cancelled, which causes a crash.

The crash can be prevented by wrapping the init call in a try-catch (OperationCancelledException), but this doesn't actually handle the cancellation. The UI will remain on whatever page it was on, rather than return to the core selection or OOBE root page.

## Checklist

This PR meets the following requirements:
- [ ] Tested and contains **NO** breaking changes or known regressions.
- [ ] Tested with the upstream branch merged in.
- [ ] Tests have been added for bug fixes / features (or this option is not applicable)
- [ ] All new code has been documented (or this option is not applicable)
- [ ] Headers have been added to all new source files (or this option is not applicable)

## Additional info

Not provided

### CODE REVIEW

File: src/Shared/AppLoadingView.xaml.cs [hunk 1]
```diff
@@ -259,7 +259,18 @@ async void OutOfBox_CoreInstanceRegistered(object? sender, CoreInstanceEventArgs
...
+                {
+                    await core.InitAsync();
+                }
+                catch (OperationCanceledException)
+                {
+                    // TODO: Does the core need to be disposed?
+                    // TODO: Navigate back to OOBE proper. As is, it just stays
+                    // on the core configuration page
+
+                    //await core.DisposeAsync();
```
Comment (yoshiask) [inline src/Shared/AppLoadingView.xaml.cs:272]: Does the core need to be disposed and/or removed from `CoreRegistry`?

File: src/Shared/AppLoadingView.xaml.cs [hunk 2]
```diff
@@ -259,7 +259,18 @@ async void OutOfBox_CoreInstanceRegistered(object? sender, CoreInstanceEventArgs
...
 
-                await core.InitAsync();
+                try
+                {
+                    await core.InitAsync();
+                }
+                catch (OperationCanceledException)
+                {
+                    // TODO: Does the core need to be disposed?
+                    // TODO: Navigate back to OOBE proper. As is, it just stays
```
Comment (yoshiask) [inline src/Shared/AppLoadingView.xaml.cs:269]: What is the proper way to navigate to the first page in the OOBE navigation stack that isn't from a core? Or should we always navigate to the core selection page?

Comment (Arlodotexe) [reply src/Shared/AppLoadingView.xaml.cs:269]: The core should be unregistered with the `coreManagementService`, and the UI will pick up on that. You should have access to an instance of this.

File: src/Shared/AppLoadingView.xaml.cs [hunk 3]
```diff
@@ -259,7 +259,16 @@ async void OutOfBox_CoreInstanceRegistered(object? sender, CoreInstanceEventArgs
 
                 core.CoreStateChanged += Core_CoreStateChanged;
 
-                await core.InitAsync();
+                try
```
Comment (Arlodotexe) [inline src/Shared/AppLoadingView.xaml.cs:262]: You'll want to do these same changes to the other non-oobe (lifetime) event handler just below this.

**Review by Arlodotexe — CHANGES_REQUESTED** [review]

File: src/Shared/AppLoadingView.xaml.cs [hunk 4]
```diff
@@ -280,7 +289,16 @@ async void Lifetime_CoreInstanceRegistered(object? sender, CoreInstanceEventArgs
...
+                {
+                    await core.InitAsync();
+                }
+                catch (OperationCanceledException)
+                {
+                    // The core initialization was cancelled.
+                    // Unregister from the core service, everything should update from there.
+                    await coreManagementService.UnregisterCoreInstanceAsync(args.InstanceId);
```
Comment (Arlodotexe) [inline src/Shared/AppLoadingView.xaml.cs:300]:
```suggestion
                    await coreManagementService.UnregisterCoreInstanceAsync(args.InstanceId);
                    return;
```

Comment (Arlodotexe) [inline src/Shared/AppLoadingView.xaml.cs:300]:
```suggestion
                    await coreManagementService.UnregisterCoreInstanceAsync(args.InstanceId);
                    return;
```

**Review by Arlodotexe — APPROVED** [review]

**Review by Arlodotexe — APPROVED** [review]

**Pull request merged**